code/game/objects/structures/tables_racks.dm

```diff
@@ -263,29 +263,54 @@
 	buildstack = /obj/item/stack/sheet/glass
 	canSmoothWith = null
 	health = 50
+	var/list/debris = list()
+
+/obj/structure/table/glass/New()
+	. = ..()
+	debris += new frame
+	debris += new /obj/item/weapon/shard
+
+/obj/structure/table/glass/Destroy()
+	for(var/i in debris)
+		qdel(i)
+	. = ..()
 
 /obj/structure/table/glass/Crossed(atom/movable/AM)
 	. = ..()
 	if(flags & NODECONSTRUCT)
 		return
+	if(!isliving(AM))
+		return
 	// Don't break if they're just flying past
 	if(AM.throwing)
-		return
+		spawn(5)

```

Comment:
Is there no way to check that this is the last step of a throw without this spawn?

Like, modify the throwing proc so taht when it unsets AM.throwing it calls throw_impact on everything in the tile.